torch/_functorch/partitioners.py

```diff
@@ -92,6 +92,8 @@ class NodeInfo:
     required_bw_nodes: OrderedSet[fx.Node]
     unclaimed_nodes: OrderedSet[fx.Node]
     fw_order: dict[fx.Node, int]
+    # Effectively maps to which of our primals are parameters
+    static_input_nodes: OrderedSet[fx.Node]
```

Comment:
Call it param_indices or something if it's just tied to parameter-ness. I'm not sure it is completely (what about buffers?)

Reply:
in theory I think this would apply to more than just parameters. If the user has a tensor that they know will be alive the entire training loop, then they can mark it with the `static_address()` API, to convey both to the partitioner that saving it is free, and to cudagraphs that we can burn in memory addresses.

Reply:
yup